### ISSUE TITLE
Search can return invalid links

### ISSUE THREAD

**Issue by markprocess** [nim-lang/nimforum#167]
https://forum.nim-lang.org/search?q=graphql

produces some links that are invalid, for instance clicking on the response from dom96 produces

    Couldn't retrieve posts.

    Please report this issue to us so we can fix it!

I can think of a few approaches to address this:

* a maintenance sql report that identifies anomalies in the db followed by correction
* an improved sql query in the search that excludes invalid matches
* filter the search results using SearchResult.isValid
* filter the search results usingSearchResult.url().isValid

where isValid doed the obvious check.

What do you think?

**Comment by Yardanico** [comment]
@markprocess that's because this GraphQL thread was deleted (probably it was marked as "deleted" but it wasn't removed from the DB)

**Comment by Yardanico** [comment]
@dom96 if you search for `graphql` you can see `NimQL - Help and Guidance Appreciated` thread in the search results, but it's removed, so this issue is about deleted threads being visible in the search results.

**Comment by dom96** [comment]
Yes, I know. I was looking for that error message. Clicking on my post gives it.

**Comment by markprocess** [comment]
FYI if the page that prompts the use to report an issue is refreshed then a 404 is generated, which is not the right thing to do. The 404 also has a page title "$title" in the browser tab.

**Comment by markprocess** [comment]
@Yardanico @dom96 

https://forum.nim-lang.org/gibberish

prompts for an issue report but it is a syntax error. Cheers, Mark.
